validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorApiChannel.java

```diff
@@ -234,6 +240,9 @@ SafeFuture<Optional<BlockContainerAndMetaData>> createUnsignedBlock(
   SafeFuture<Optional<Attestation>> createAggregate(
       UInt64 slot, Bytes32 attestationHashTreeRoot, Optional<UInt64> committeeIndex);
 
+  SafeFuture<Optional<Attestation>> createAggregateV2(
+      UInt64 slot, Bytes32 attestationHashTreeRoot, UInt64 committeeIndex);
+
```

Comment:
I see your point introducing a V2 here, but maybe is better to revert the original `createAggregate` removing the `Optional` and add a `createAggregate` with the `committeeIndex`. So we can avoid polluting the channel with API versioning (there are none here) and still deprecate the old one. WDYT?

Reply:
And what about the remote API call which in the `createAggregateV2` uses the new `GetAggregateAttestationV2` endpoint? That was the main purpose of this versioning. I thought we could completely remove the old `createAggregate` logic once the V2 endpoint is tested and confirmed (we could still add a fallback to the V1 endpoint if we don't want to completely remove it)...

Reply:
As discussed, I kept the validator channel interface as is and put the api versioning logic in the remote api handler. Let me know what you think about it now